src/DTF/Libraries/WindowsInstaller.Package/InstallPackage.cs

```diff
@@ -202,6 +202,64 @@ private bool IsMergeModule()
     private delegate void ProcessFilesOnOneMediaDiskHandler(string mediaCab,
         InstallPathMap compressedFileMap, InstallPathMap uncompressedFileMap);
 
+    /// <summary>
+    /// Provides a key (filename) / value (full path) pair where the Binary files can be worked with
+    /// </summary>
+    /// <param name="path">The path to search for files in.  This is NOT recursive.</param>
+    /// <param name="names">The explicit names to search for.  All will be returned if nothing specified.</param>
+    /// <returns>A <see cref="M:System.Colletions.Generic.IDictionary`1{string,string}"/> of the filename/fullpath key value pairs of the directory given.</returns>
+    public virtual IDictionary<string, string> GetFilePaths(string path, ICollection<string> names = null)
+    {
+        IDictionary<string, string> retval = new Dictionary<string, string>(100);
+
+        foreach(FileInfo fi in new DirectoryInfo(path).GetFiles())
+        {
+            if (null != names && !names.Contains(fi.Name)) continue;
+
+            retval.Add(fi.Name, fi.FullName);
+        }
+
+        return retval;
+    }
+
+    /// <summary>
+    /// Extracts binary data from the `Binary` or `Icon` tables to the designated path.
+    /// </summary>
+    /// <param name="names">The names of the rows to be picked.  If null then all rows will be returned.  If none are matched then none are created.</param>
+    /// <param name="tableName">The name of the table to extract binary data from.  Valid values are "Binary" and "Icon" or a custom table with Name and Data columns of type string and stream respectively.</param>
+    /// <param name="path">The path to extract the files to.  The path must exist before calling this method.</param>
+    [SuppressMessage("Microsoft.Globalization", "CA1308:NormalizeStringsToUppercase")]
+    public virtual void ExtractFilesInBinaryTable(ICollection<string> names, string tableName, string path)
```

Comment:
In DTF, I'd want something more generic, so it didn't require the exact same column definitions. (i.e., pass in the column).